### ISSUE TITLE
Todos become orphaned when source file is moved or deleted

### ISSUE THREAD

**Issue by scalvert** [ember-template-lint/ember-template-lint#2010]
Currently, when processing files for `--clean-todo`, we iterate and operate on individual files. Due to this, we potentially orphan todo files that are no longer valid simply because we don't operate on them.

Examples:

Operating on all files:

```shell
ember-template-lint . --clean-todo
```

will operate on all files, but only those that exist in the directory at that point in time (which will exclude specific files that have been deleted). This will naturally exclude processing of todos that don't have an associated source file.

Operating on specific files

```shell
ember-template-lint app/templates/application.hbs --clean-todo
```

will operate only on that single file, obviously not operating on any files not in this list.

Steps to reproduce:

- create a template with an error
- generate todo for template
- delete template
- run `--clean-todo` or `--update-todo` against your whole working dir and expect that todo file is deleted

Currently, the todo file will not be deleted (as again, we only operate on single files).

**Comment by MelSumner** [comment]
@scalvert do you think this is a duplicate of https://github.com/ember-template-lint/ember-template-lint/issues/1969 ?

**Comment by scalvert** [comment]
@MelSumner ya it sounds the same. Should we move the contents over to that other ticket?

**Comment by MelSumner** [comment]
@scalvert yes, moved. 

Closing this issue as duplicate of #1969
